src/core/execute.c

```diff
@@ -237,38 +237,48 @@ static bool needs_cgroup_namespace(ProtectControlGroups i) {
         return IN_SET(i, PROTECT_CONTROL_GROUPS_PRIVATE, PROTECT_CONTROL_GROUPS_STRICT);
 }
 
-ProtectControlGroups exec_get_protect_control_groups(const ExecContext *context) {
+ProtectControlGroups exec_get_protect_control_groups(const ExecContext *context, const ExecParameters *params) {
         assert(context);
 
+        if (!needs_cgroup_namespace(context->protect_control_groups))
+                return context->protect_control_groups;
+
+        /* CGroup namespaces don't really make sense for control processes and can't really be supported as
+         * (some) control processes need to be spawned directly into a subcgroup to avoid violating the
+         * "no inner processes" rule of cgroupv2, so don't do any cgroup namespacing for control processes. */
```

Comment:
hmm, not sure i follow here? see exec_params_get_cgroup_path(). it inserts a .control/ cgroup to address this issue already?

Reply:
See above, I think spawning control processes in a different cgroup namespace root from the main process isn't useful at all for anyone.